File: 代码实现/block强弱引用/block强弱引用/ViewController.m
//
//  ViewController.m
//  block强弱引用
//
//  Created by kevin on 2017/7/6.
//  Copyright © 2017年 kevin. All rights reserved.
//

#import "ViewController.h"
#import "TestObject.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    TestObject *obj = [TestObject new];//在创建的时候设定好要执行的动作，
    obj.block();//一秒后打印下obj
    
    
    //1秒后还在不在，此时，obj已经被释放
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 代码实现/测试区域/unitTest/unitTest/ViewController.swift
//
//  ViewController.swift
//  unitTest
//
//  Created by 张俊凯 on 2017/5/10.
//  Copyright © 2017年 张俊凯. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        print("自定义测试")
        let a = 3
        //assert(a == 0)
        
        //闭包的使用
        typealias add = (NSInteger,NSInteger)->(NSInteger)
        var adda:add
        adda = {
            (_ num1:NSInteger,_ num2:NSInteger) -> (NSInteger) in
            return num1 + num2
        }
        print(adda(2,3))
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

